src/NServiceBus.Transport.SQS/DelayedMessagesPump.cs

```diff
@@ -292,67 +299,85 @@ async Task SendDelayedMessagesInBatches(BatchEntry<SqsReceivedDelayedMessage> ba
-                    {
-                        VisibilityTimeout = 0
-                    });
-                }
-
-                if (changeVisibilityBatchRequestEntries != null)
+            // deliberately fire&forget because we treat this as a best effort
+            _ = Task.Run(
+                async () =>
                 {
-                    if (Logger.IsDebugEnabled)
+                    try
                     {
-                        var message = batch.PreparedMessagesBydId.Values.First();
-
-                        Logger.Debug($"Changing delayed message visibility for batch '{batchNumber}/{totalBatches}' with message ids '{string.Join(", ", batch.PreparedMessagesBydId.Values.Select(v => v.MessageId))}' for destination {message.Destination}");
+                        await ChangeVisibilityOfDelayedMessagesThatFailedBatchDeliveryInBatches(batch, result, batchNumber, totalBatches, cancellationToken).ConfigureAwait(false);
                     }
-
-                    var changeVisibilityResult = await sqsClient.ChangeMessageVisibilityBatchAsync(new ChangeMessageVisibilityBatchRequest(delayedDeliveryQueueUrl, changeVisibilityBatchRequestEntries), cancellationToken)
-                        .ConfigureAwait(false);
-
-                    if (Logger.IsDebugEnabled)
+                    catch (OperationCanceledException ex)
                     {
-                        var message = batch.PreparedMessagesBydId.Values.First();
-
-                        Logger.Debug($"Changed delayed message visibility for batch '{batchNumber}/{totalBatches}' with message ids '{string.Join(", ", batch.PreparedMessagesBydId.Values.Select(v => v.MessageId))}' for destination {message.Destination}");
+                        if (cancellationToken.IsCancellationRequested)
+                        {
+                            Logger.Debug("Changing visibility of delayed messages that failed batch delivery cancelled.", ex);
+                        }
+                        else
+                        {
+                            Logger.Warn("OperationCanceledException thrown.", ex);
+                        }
                     }
-
-                    if (Logger.IsDebugEnabled && changeVisibilityResult.Failed.Count > 0)
+                    catch (Exception ex)
                     {
                         var builder = new StringBuilder();
-                        foreach (var failed in changeVisibilityResult.Failed)
+                        foreach (var failed in result.Failed)
                         {
                             builder.AppendLine($"{failed.Id}: {failed.Message} | {failed.Code} | {failed.SenderFault}");
                         }
 
-                        Logger.Debug($"Changing visibility failed for {builder}");
+                        Logger.Error($"Error changing visibility for {builder}", ex);
```

Comment:
I assume this was changed to ERROR on purpose? any risk we're crying wolf in logs now?

Reply:
Yeah, it was on purpose. It seemed off to be logging an error at debug level. Anyone running at anything less than info would never know there is any problem here.

However, given the comment above says:

> // deliberately fire&forget because we treat this as a best effort

Perhaps this should be only a warning?

Reply:
WARN sounds like a good trade off